sql/src/test/java/io/crate/integrationtests/SnapshotIntegrationTest.java

```diff
@@ -119,4 +122,113 @@ public void testDropSnapshotUnknownRepository() throws Exception {
+                {3L, "baz", "1989-11-09", "Reden ist Schweigen. Silber ist Gold."},
+        });
+        execute("REFRESH TABLE " + tableName);
+    }
+
+    @Test
+    public void testCreateSnapshot() throws Exception {
+        createTable("backmeup", false);
+
+        execute("CREATE SNAPSHOT " + snapshotName() + " TABLE backmeup WITH (wait_for_completion=true)");
+        assertThat(response.rowCount(), is(1L));
+
+        execute("select name, \"repository\", concrete_indices, state from sys.snapshots");
+        assertThat(TestingHelpers.printedTable(response.rows()),
+                is("my_snapshot| my_repo| [backmeup]| SUCCESS\n"));
+    }
+
+    @Test
+    public void testCreateSnapshotFromPartition() throws Exception {
+        createTable("custom.backmeup", true);
+
+        execute("CREATE SNAPSHOT " + snapshotName() + " TABLE custom.backmeup PARTITION (date='1970-01-01')  WITH (wait_for_completion=true)");
+        assertThat(response.rowCount(), is(1L));
+
+        execute("select name, \"repository\", concrete_indices, state from sys.snapshots");
+        assertThat(TestingHelpers.printedTable(response.rows()),
+                is("my_snapshot| my_repo| [custom..partitioned.backmeup.04130]| SUCCESS\n"));
+    }
+
+    @Test
+    public void testCreateExistingSnapshot() throws Exception {
+         createTable("backmeup", randomBoolean());
+
+        execute("CREATE SNAPSHOT " + snapshotName() + " ALL WITH (wait_for_completion=true)");
+        assertThat(response.rowCount(), is(1L));
+
+        expectedException.expect(SQLActionException.class);
+        expectedException.expectMessage("Snapshot \"my_repo\".\"my_snapshot\" already exists");
+        execute("CREATE SNAPSHOT my_repo.my_snapshot ALL WITH (wait_for_completion=true)");
+    }
+
+    @Test
+    public void testCreateSnapshotUnknownRepo() throws Exception {
+        expectedException.expect(SQLActionException.class);
+        expectedException.expectMessage("Repository 'unknown_repo' unknown");
+        execute("CREATE SNAPSHOT unknown_repo.my_snapshot ALL WITH (wait_for_completion=true)");
+    }
+
+    @Test
+    public void testInvalidSnapshotName() throws Exception {
+
+        expectedException.expect(SQLActionException.class);
+        expectedException.expectMessage("Invalid snapshot name [MY_UPPER_SNAPSHOT], must be lowercase");
+        execute("CREATE SNAPSHOT my_repo.\"MY_UPPER_SNAPSHOT\" ALL WITH (wait_for_completion=true)");
+    }
+
+    @Test
+    public void testCreateNotPartialSnapshotFails() throws Exception {
+        execute("set global cluster.routing.allocation.enable=none");
+        execute("CREATE TABLE backmeup (" +
+                "  id long primary key, " +
+                "  name string" +
+                ") with (number_of_replicas=0)");
+
+        expectedException.expect(SQLActionException.class);
+        expectedException.expectMessage("[my_repo:my_snapshot] Indices don't have primary shards [backmeup]");
```

Comment:
pls provide better (crate) message. (tables instead of indices..., snapshot name)